angular-ngrx-scss/src/app/state/auth/auth.selectors.ts

```diff
@@ -9,3 +9,13 @@ export const selectIsAuthenticated = createSelector(
   selectAuthState,
   (state: AuthState) => state.isAuthenticated,
 );
+
+export const selectAuthUserAvatar = createSelector(
+  selectAuthState,
+  (state: AuthState) => state.authUser.avatar,
+);
+
+export const selectAuthUserName = createSelector(
+  selectAuthState,
+  (state: AuthState) => state.authUser.username,
+);
```

Comment:
I suggest using more selectors composition. This way you reuse some of the logic as well as can get some performance benefits. Additionally you don't need to explicitly provide types for the `state` etc. `createSelector` is strictly typed and all those types will be inferred

```suggestion

const selectAuthUser = createSelector(selectAuthState, state => state.authState)

export const selectAuthUserAvatar = createSelector(
  selectAuthUser,
  authUser => authUser.avatar,
);

export const selectAuthUserName = createSelector(
  selectAuthUser,
  authUser => authUser.username,
);
```

Reply:
Appreciate this!